src/__test__/extra/typedEvents-more.test.ts

```diff
@@ -0,0 +1,296 @@
+        return {
+          id: event.data.shoppingCartId,
+          clientId: event.data.clientId,
+          openedAt: new Date(event.data.openedAt),
+          productItems: new Map(),
+          status: ShoppingCartStatus.Opened,
+        };
+      }
+
+      if (currentState == null) throw ShoppingCartErrors.CART_NOT_FOUND;
+
+      switch (event.type) {
+        case "product-item-added-to-shopping-cart":
+          return {
+            ...currentState,
+            productItems: addProductItem(
+              currentState.productItems,
+              event.data.productItem
+            ),
+          };
+        case "product-item-removed-from-shopping-cart":
+          return {
+            ...currentState,
+            productItems: removeProductItem(
+              currentState.productItems,
+              event.data.productItem
+            ),
+          };
+        case "shopping-cart-confirmed":
+          return {
+            ...currentState,
+            status: ShoppingCartStatus.Confirmed,
+            confirmedAt: new Date(event.data.confirmedAt),
+          };
+        default: {
+          const _: never = event;
+          throw ShoppingCartErrors.UNKNOWN_EVENT_TYPE;
+        }
+      }
+    });
+
+    const enum ProductsIds {
+      T_SHIRT = "team-building-excercise-2022",
+      SHOES = "air-jordan",
+    }
+
+    const clientId = "client-123";
+    const shoppingCartId = "cart-456";
+    const events: ShoppingCartEvent[] = [
+      {
+        type: "shopping-cart-opened",
+        data: {
+          shoppingCartId,
+          clientId,
+          openedAt: new Date().toJSON(),
+        },
+      },
+      {
+        type: "product-item-added-to-shopping-cart",
+        data: {
+          shoppingCartId,
+          productItem: {
+            productId: ProductsIds.SHOES,
+            quantity: 100,
+          },
+        },
+      },
+      {
+        type: "product-item-added-to-shopping-cart",
+        data: {
+          shoppingCartId,
+          productItem: {
+            productId: ProductsIds.T_SHIRT,
+            quantity: 1,
+          },
+        },
+      },
+      {
+        type: "product-item-removed-from-shopping-cart",
+        data: {
+          shoppingCartId,
+          productItem: {
+            productId: ProductsIds.SHOES,
+            quantity: 100,
+          },
+        },
+      },
+      {
+        type: "product-item-added-to-shopping-cart",
+        data: {
+          shoppingCartId,
+          productItem: {
+            productId: ProductsIds.T_SHIRT,
+            quantity: 1,
+          },
+        },
+      },
+      {
+        type: "shopping-cart-confirmed",
+        data: {
+          shoppingCartId,
+          confirmedAt: new Date().toJSON(),
+        },
+      },
+    ];
+
+    const jsonEvents = events.map((e) => jsonEvent<ShoppingCartEvent>(e));
+
+    await client.appendToStream<ShoppingCartEvent>(
+      `${clientId}-${shoppingCartId}`,
```

Comment:
**FIX:** Typically, you should have the stream name starting with the category name, then streams 
```suggestion
      `$shoppingcart-{clientId}_${shoppingCartId}`,
```